java-checks/src/main/java/org/sonar/java/checks/CastArithmeticOperandCheck.java

```diff
@@ -77,6 +77,11 @@ public void visitNode(Tree tree) {
         if (returnType != null && isVarTypeErrorProne(returnType)) {
           methodTree.accept(new ReturnStatementVisitor(returnType));
         }
+      } else if (tree.is(Tree.Kind.DIVIDE)) {
```

Comment:
I think that now that we have a 5th subscribed tree, it would be now more elegant to use a switch to cover all kinds. In a second commit.